testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -287,6 +290,316 @@ def test_write_none(self):
             w.write(None)
 
 
+class BaseTimestepTest(TestCase):
+    """Test all the base functionality of a Timestep
+
+    All Timesteps must pass these tests!
+
+    These test the Timestep independent of the Reader which it
+    comes into contact with.  Failures here are the Timesteps fault.
+    """
+    # define the class made in test
+    Timestep = mda.coordinates.base.Timestep
+    name = "base"  # for error messages only
+    size = 10  # size of arrays, 10 is enough to allow slicing etc
+    # each coord is unique
+    refpos = np.arange(size * 3, dtype=np.float32).reshape(size, 3)
+    has_box = True
+    set_box = True  # whether you can set dimensions info.
+    # If you can set box, what the underlying unitcell should be
+    # if dimensions are:
+    newbox = np.array([10., 11., 12., 90., 90., 90.])
+    unitcell = np.array([10., 11., 12., 90., 90., 90.])
+    ref_volume = 1320.  # what the volume is after setting newbox
+
+    def setUp(self):
+        self.ts = self.Timestep(self.size)
+        self.ts.frame += 1
+        self.ts.positions = self.refpos
+
+    def tearDown(self):
+        del self.ts
+
+    def test_getitem(self):
+        assert_equal(self.ts[1], self.refpos[1])
+
+    def test_getitem_neg(self):
+        assert_equal(self.ts[-1], self.refpos[-1])
+
+    def test_getitem_neg_IE(self):
+        assert_raises(IndexError, self.ts.__getitem__, -(self.size + 1))
+
+    def test_getitem_pos_IE(self):
+        assert_raises(IndexError, self.ts.__getitem__, (self.size + 1))
+
+    def test_getitem_slice(self):
+        assert_equal(len(self.ts[:2]), len(self.refpos[:2]))
+        assert_allclose(self.ts[:2], self.refpos[:2])
+
+    def test_getitem_slice2(self):
+        assert_equal(len(self.ts[1::2]), len(self.refpos[1::2]))
+        assert_allclose(self.ts[1::2], self.refpos[1::2])
+
+    def test_getitem_ndarray(self):
+        sel = np.array([0, 1, 4])
+        assert_equal(len(self.ts[sel]), len(self.refpos[sel]))
+        assert_allclose(self.ts[sel], self.refpos[sel])
+
+    def test_getitem_TE(self):
+        assert_raises(TypeError, self.ts.__getitem__, 'string')
+
+    def test_len(self):
+        assert_equal(len(self.ts), self.size)
+
+    def test_iter(self):
+        for a, b in zip(self.ts, self.refpos):
+            assert_allclose(a, b)
```

Comment:
This is slicing a `Timestep` so it just returns positions for now, not another Timestep